### PR TITLE
remove-un-used-Evaluate-in-FailMacros

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Review by MiYanni — APPROVED** [review]
The IList change seems innocuous so I'll approve.

File: test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/TemplateConfigTests/SymbolConfigTests.cs [hunk 1]
```diff
@@ -670,7 +670,7 @@ public void ObjectConfigNameSymbolWithIdentityFormAndAddIdentityFalseRetainsConf
 
             ParameterSymbol? nameSymbol = symbolInfo as ParameterSymbol;
             Assert.NotNull(nameSymbol);
-            var configuredValueFormNames = nameSymbol!.Forms.GlobalForms;
```
[the remark this answers is not in the document]
Comment (MiYanni) [reply test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/TemplateConfigTests/SymbolConfigTests.cs:673]: GlobalForms is an IReadOnlyList and it is used for read-only purposes. So, I'm not sure what making it a List boxed as an IList provides.

[the remark this answers is not in the document]
Comment (SimonCropp) [reply test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/TemplateConfigTests/SymbolConfigTests.cs:620]: treating a list as an IList is not boxing

note i just moved the multiple duplicated `.ToList();` below up a single variable

[the remark this answers is not in the document]
Comment (SimonCropp) [reply test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/TemplateConfigTests/SymbolConfigTests.cs:620]: but either way. these changes should not have been in this PR. i will revert this part

**Review by Forgind — APPROVED** [review]

**Pull request merged**